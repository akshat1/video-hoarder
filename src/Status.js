/**
 * @enum {string} Status
 */
export default {
  Pending: 'pending',
  Running: 'running',
  Failed: 'failed',
  Succeeded: 'succeeded'
};
