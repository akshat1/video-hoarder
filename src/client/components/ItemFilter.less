@import (reference) "../style/vars.less";

.wrapper {
  max-width: 20rem;
  button {
    padding: 0.5rem;
    text-transform: capitalize;
  }
}

.desktop {
  .desktop-only;
}

.mobile {
  .mobile-only;
}

select, option {
  text-transform: capitalize;
}
