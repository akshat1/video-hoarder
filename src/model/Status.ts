export enum Status {
  Failed = "failed",
  Paused = "paused",
  Pending = "pending",
  Running = "running",
  Succeeded = "succeeded",
}

export enum StatusFilterValue {
  All = "all",
  Failed = "failed",
  Paused = "paused",
  Pending = "pending",
  Running = "running",
  Succeeded = "succeeded",
}


export const hasStarted = (status: Status): boolean =>
  status === Status.Running || status === Status.Failed || status === Status.Succeeded || status === Status.Paused;

export const hasConcluded = (status: Status): boolean =>
  status === Status.Failed || status === Status.Succeeded;

export const isPending = (status: Status): boolean =>
  status === Status.Pending;
