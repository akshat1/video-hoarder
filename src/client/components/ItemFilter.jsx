/**
 * This component renders a drop-down on mobile, or a button group on desktop, letting the user
 * pick one of the item statuses.
 */
import { Status } from "../../Status";
import ButtonGrid from "./ButtonGrid.jsx";
import Style from "./ItemFilter.less";
import PropTypes from "prop-types";
import React from "react";

const ItemFilter = ({ value, onChange }) => {
  const options = Object.values(Status).sort().map(status => ({
    label: status,
    value: status,
    selected: status === value,
  }));

  return (
    <div className={Style.wrapper}>
      <div className={Style.mobile}>
        <select value={value} onChange={e => onChange(e.currentTarget.value)}>
          <For each="option" of={options}>
            <option
              value={option.value}
              key={option.value}
            >
              {option.label}
            </option>
          </For>
        </select>
      </div>
      <div className={Style.desktop}>
        <ButtonGrid
          onChange={onChange}
          options={options}
        />
      </div>
    </div>
  );
};

ItemFilter.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.oneOf(Object.values(Status)),
};
export default ItemFilter;
