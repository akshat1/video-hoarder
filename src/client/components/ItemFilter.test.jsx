import React from 'react';
import { shallow } from 'enzyme';
import ItemFilter from './ItemFilter.jsx';
import { Status } from '../../Status';

describe('components/ItemFilter', () => {
  test('should render and match snapshot', () => {
    Object.values(Status)
      .map(status =>
        expect(shallow(<ItemFilter onChange={() => 0} value={status}/>)).toMatchSnapshot()
      );
  });

  test('calls onChange', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<ItemFilter onChange={onChange} value={Status.Failed}/>);
    const select = wrapper.find('select');
    select.simulate(
      'change',
      {
        currentTarget: { value: Status.Pending },
      },
    );
    expect(onChange).toHaveBeenCalledWith(Status.Pending);
  });
});
