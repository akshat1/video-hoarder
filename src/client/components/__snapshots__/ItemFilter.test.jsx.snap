// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ItemFilter should render and match snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="mobile"
  >
    <select
      onChange={[Function]}
      value="pending"
    >
      <option
        key="failed"
        value="failed"
      >
        failed
      </option>
      <option
        key="pending"
        value="pending"
      >
        pending
      </option>
      <option
        key="running"
        value="running"
      >
        running
      </option>
      <option
        key="succeeded"
        value="succeeded"
      >
        succeeded
      </option>
    </select>
  </div>
  <div
    className="desktop"
  >
    <ButtonGrid
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "failed",
            "selected": false,
            "value": "failed",
          },
          Object {
            "label": "pending",
            "selected": true,
            "value": "pending",
          },
          Object {
            "label": "running",
            "selected": false,
            "value": "running",
          },
          Object {
            "label": "succeeded",
            "selected": false,
            "value": "succeeded",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`components/ItemFilter should render and match snapshot 2`] = `
<div
  className="wrapper"
>
  <div
    className="mobile"
  >
    <select
      onChange={[Function]}
      value="running"
    >
      <option
        key="failed"
        value="failed"
      >
        failed
      </option>
      <option
        key="pending"
        value="pending"
      >
        pending
      </option>
      <option
        key="running"
        value="running"
      >
        running
      </option>
      <option
        key="succeeded"
        value="succeeded"
      >
        succeeded
      </option>
    </select>
  </div>
  <div
    className="desktop"
  >
    <ButtonGrid
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "failed",
            "selected": false,
            "value": "failed",
          },
          Object {
            "label": "pending",
            "selected": false,
            "value": "pending",
          },
          Object {
            "label": "running",
            "selected": true,
            "value": "running",
          },
          Object {
            "label": "succeeded",
            "selected": false,
            "value": "succeeded",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`components/ItemFilter should render and match snapshot 3`] = `
<div
  className="wrapper"
>
  <div
    className="mobile"
  >
    <select
      onChange={[Function]}
      value="failed"
    >
      <option
        key="failed"
        value="failed"
      >
        failed
      </option>
      <option
        key="pending"
        value="pending"
      >
        pending
      </option>
      <option
        key="running"
        value="running"
      >
        running
      </option>
      <option
        key="succeeded"
        value="succeeded"
      >
        succeeded
      </option>
    </select>
  </div>
  <div
    className="desktop"
  >
    <ButtonGrid
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "failed",
            "selected": true,
            "value": "failed",
          },
          Object {
            "label": "pending",
            "selected": false,
            "value": "pending",
          },
          Object {
            "label": "running",
            "selected": false,
            "value": "running",
          },
          Object {
            "label": "succeeded",
            "selected": false,
            "value": "succeeded",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`components/ItemFilter should render and match snapshot 4`] = `
<div
  className="wrapper"
>
  <div
    className="mobile"
  >
    <select
      onChange={[Function]}
      value="succeeded"
    >
      <option
        key="failed"
        value="failed"
      >
        failed
      </option>
      <option
        key="pending"
        value="pending"
      >
        pending
      </option>
      <option
        key="running"
        value="running"
      >
        running
      </option>
      <option
        key="succeeded"
        value="succeeded"
      >
        succeeded
      </option>
    </select>
  </div>
  <div
    className="desktop"
  >
    <ButtonGrid
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "failed",
            "selected": false,
            "value": "failed",
          },
          Object {
            "label": "pending",
            "selected": false,
            "value": "pending",
          },
          Object {
            "label": "running",
            "selected": false,
            "value": "running",
          },
          Object {
            "label": "succeeded",
            "selected": true,
            "value": "succeeded",
          },
        ]
      }
    />
  </div>
</div>
`;
